internal/controllers/utils/utils.go

```diff
@@ -602,6 +601,20 @@ func ExtractTemplateDataFromConfigMap[T any](ctx context.Context, c client.Clien
 	return validData, nil
 }
 
+// ExtractTimeoutConfigFromConfigMap extracts the timeout config from the ConfigMap by key if exits,
+// converting it from integer to time.Duration. Returns an error if the value is not a valid integer.
+func ExtractTimeoutConfigFromConfigMap(cm *corev1.ConfigMap, key string) (time.Duration, error) {
+	if timeoutStr, exists := cm.Data[key]; exists {
+		timeoutInt, err := strconv.Atoi(timeoutStr)
+		if err != nil {
+			return 0, NewInputError("the value of key %s from ConfigMap %s is not an integer: %v", key, cm.GetName(), err)
+		}
+		return time.Duration(timeoutInt) * time.Minute, nil
```

Comment:
In my opinion, we should just return `timeoutInt` here, no conversion, and leave it to the `TimeoutExceeded` function.

Reply:
The type of parameters in the timeouts is time.Duration, same for the default values in constants.go. I didn't use int because I feel a dedicated type time.Duration represents the intervals more clearly
```
	DefaultHardwareProvisioningTimeout = 90 * time.Minute
	DefaultClusterProvisioningTimeout  = 90 * time.Minute
	DefaultClusterConfigurationTimeout = 30 * time.Minute
```